examples/yc-batch-company-employees/src/lib/services/linkedin-extractor.service.ts

```diff
@@ -123,110 +87,96 @@ export class LinkedInExtractorService {
-
-          return employeesListUrls;
-        } catch (error: any) {
-          this.log
-            .withError(error?.message ?? error)
-            .withMetadata({
-              sessionId,
-              windowId,
-            })
-            .error(`Error extracting employees list URL for company LinkedIn profile: ${url}`);
-          return null;
-        }
+    companyLinkedInProfileUrls: BatchOperationUrl[];
+    profileId: string;
+  }): Promise<BatchOperationUrl[]> {
+    this.log.info("Attempting to get the list of employees for the companies");
+
+    const getEmployeesListUrl = async (input: BatchOperationInput): Promise<BatchOperationResponse<string>> => {
+      const scrapedContent = await this.airtop.client.windows.scrapeContent(input.sessionId, input.windowId);
+
+      const url = this.extractEmployeeListUrl(scrapedContent.data.modelResponse.scrapedContent.text);
+
+      if (!url) {
+        throw new Error("No employees list URL found");
+      }
+
+      return {
+        data: url,
+      };
+    };
+
+    const handleError = async ({ error, operationUrls, liveViewUrl }: BatchOperationError) => {
+      this.log
+        .withError(error)
+        .withMetadata({
+          liveViewUrl,
+          operationUrls,
+        })
+        .error("Error extracting employees list URL for company LinkedIn profile.");
+    };
+
+    const employeesListUrls = await this.airtop.client.batchOperate(companyLinkedInProfileUrls, getEmployeesListUrl, {
+      onError: handleError,
+      sessionConfig: {
+        baseProfileId: profileId,
       },
     });
 
-    await this.airtop.terminateAllWindows();
+    this.log
+      .withMetadata({
+        employeesListUrls,
+      })
+      .info("Successfully fetched employee list URLs for the companies");
 
     // Filter out any null values and remove duplicates
-    return [...new Set(employeesListUrls.filter((url) => url !== null))];
+    return [...new Set(employeesListUrls.filter((url) => url !== null).map((url) => ({ url })))];
   }
 
   /**
    * Gets the LinkedIn employees profile URLs for a list of LinkedIn employees list URLs
    * @param employeesListUrls - The list of LinkedIn employees list URLs
    * @param sessionId - The ID of the session
-   * @param parallelism - The number of parallel requests to make
    * @returns The list of LinkedIn employees profile URLs
    */
   async getEmployeesProfileUrls({
     employeesListUrls,
-    sessionId,
-    parallelism,
-  }: { employeesListUrls: string[]; sessionId: string; parallelism: number }): Promise<string[]> {
-    this.log
-      .withMetadata({
-        parallelism,
+    profileId,
+  }: { employeesListUrls: BatchOperationUrl[]; profileId: string }): Promise<string[]> {
+    this.log.info("Initiating extraction of employee's profile URLs for the employees");
+
+    const getEmployeeProfileUrl = async (input: BatchOperationInput): Promise<BatchOperationResponse<string[]>> => {
```

Comment:
Thanks for the implementation of the batchOperations in this PR, the code is very simply to understand!